FAO Publications.js

```diff
@@ -0,0 +1,489 @@
+			if (key.includes('seriesNumber') === true) {
+				newItem.seriesNumber = metaResult[0].toUpperCase() + metaResult.slice(1);
+			}
+			//use confCode as 'Proceedings Title' in Zotero.
+			if (key.includes('confCode') === true) {
+				newItem.publicationTitle = metaResult;
+			}
+			//conferenceName: save for later conditions.
+			if (key.includes('conference') === true) {
+				var conferenceWeb = metaResult[0];
+			}
+			if (key.includes('session') === true) {
+				var sessionWeb = metaResult;
+			}
+		}
+
+		//If there's no publisher, use 'FAO' as publisher.
+		if (newItem.publisher == null) {
+			newItem.publisher = 'FAO';
+		}
+		//Write corporate author; if no individual or corporate author, use 'FAO' as author.
+		if (newItem.creators.length == 0) {
+			newItem.creators.push({creatorType: 'author'});
+			if (corpAuthorWeb && officeWeb) {
+				newItem.creators.push({lastName: corpAuthorWeb + ', ' + officeWeb,});
+			} else if (corpAuthorWeb && !officeWeb) {
+				newItem.creators.push({lastName: corpAuthorWeb});
+			} else if (!corpAuthorWeb && officeWeb) {
+				newItem.creators.push({lastName: officeWeb});
+			} else if (!corpAuthorWeb && !officeWeb) {
+				newItem.creators.push({lastName: 'FAO'});
+			}
+		}
+		//Write conferenceName
+		if (conferenceWeb && sessionWeb) {
+			newItem.conferenceName = conferenceWeb + ' ' + sessionWeb;
+		} else if (!sessionWeb) {
+			newItem.conferenceName = conferenceWeb;
+		} else {
+			newItem.conferenceName = sessionWeb;
+		}
+		//If conference and/or session exists in document page, the itemType is 'conferencePaper'; otherwise it's 'book'. 
+		if (conferenceWeb || sessionWeb) {
+			newItem.itemType = 'conferencePaper';
+		} else {
+			newItem.itemType = 'book';
+		}
+		//********** End dynamic-location variables **********
+
+
+		//********** Others **********
+		//Place: not shown in page. Just differentiate Bangkok / Rome.
+		if (newItem.publisher.includes('Regional Office for Asia and the Pacific') === true) {
+			newItem.place = 'Bangkok';
+		} else {
+			newItem.place = 'Rome';
+		}
+	}
+	newItem.complete();
+}
+
+
+//get items from a multiple-item page
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = ZU.xpath(doc, '//*[@class="item-image"]');
+	for (var i = 0; i < rows.length; i++) {
+		var href = rows[i].href
+		var title = ZU.trimInternal(rows[i].text);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			Z.debug(items);
+			if (!items) {
+				return true;
+			}
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+}
+
+
+//********** BEGIN TEST CASES **********
+var testCases = [
```

Comment:
How did you do the test cases? It looks that you did not use [Scaffold](https://github.com/zotero/scaffold) for that...